src/core/reference/src/op/fake_convert.cpp

```diff
@@ -0,0 +1,163 @@
+            /* 0 00000 0010000000, find grs patterns */
+            // 0xx - do nothing
+            // 100 - this is a tie : round up if the mantissa's bit just before G is 1, else do nothing
+            // 101, 110, 111 - round up > 0x0080
+            val_bit_repr += (((rnmask > 0x0080) || (rnmask_tie == rne_tie)) << lshift);
+        }
+        val_bit_repr &= mask_mant; /* truncation */
+        out_u[i] = val_bit_repr;
+    }
+}
+
+/**
+ * @brief Emulation of conversion fp16 value to f8e4m3 format
+ *
+ * @param arg_f       Pointer to the input data.
+ * @param out_f       Pointer to the otuput data.
+ * @param count     Number of elements in the data input.
+ *
+ * Exponent denormal values 0 -7
+ * Exponent normal values 1..15 -6..8 (7 - exponent)
+ * Exponent NaN values 15 8
+ *
+ */
+void emulate_f8e4m3_on_fp16(const float16* arg_f, float16* out_f, size_t count) {
+    const auto arg_u = reinterpret_cast<const uint16_t*>(arg_f);
+    auto out_u = reinterpret_cast<uint16_t*>(out_f);
+    uint16_t val_bit_repr;
+
+    constexpr auto use_clamp = true;
+    constexpr auto exp_bits = 5;
+    constexpr auto mbits = 9;
+    constexpr auto non_mant_bits = exp_bits + 1;  /// exponent + sign
+    constexpr auto lshift = 10 - (mbits - non_mant_bits);
+    constexpr auto fp16_exp_bias = 15;
+    constexpr auto f8e4m3_min_val = 0.001953125f;  /// 2**-9
+
+    constexpr uint16_t mask_mant = static_cast<uint16_t>(0xFFFF << lshift);  /// 1111111111111111 -> 1 11111 1111000000
+    constexpr uint16_t grs_bitmask = 0x007F;  /// 0 00000 0001111111, grs denotes guard, round, sticky bits
+    constexpr uint16_t rne_tie = 0x00C0;      /// 0 00000 0011000000, rne denotes round to nearest even
+    constexpr uint16_t rne_mask = 1;
+    constexpr uint16_t fp16_inf = 0x7C00;
+
+    for (size_t i = 0; i < count; ++i) {
+        val_bit_repr = arg_u[i];
+        /* flush values below 1-4-3 (offset=4) subnormal range to zero */
+        if (std::abs(static_cast<float>(arg_f[i])) < f8e4m3_min_val) {
+            val_bit_repr = 0;
+        }
+
+        short exp_h = static_cast<short>((val_bit_repr & fp16_inf) >> 10) -
+                      fp16_exp_bias;  /// 0111110000000000 -> 0000000000011111 - 15, biased exponent for fp16
+        const short sign_h = (val_bit_repr & 0x8000);  /// & 1 00000 0000000000
+        short mantissa_h = (val_bit_repr & 0x03FF);    /// & 0 00000 1111111111
+        ///(val_bit_repr && 0111111111111111) < 0 10010 1110000000 (19326)
+        const bool can_round = ((val_bit_repr & 0x7FFF) < 0b101111110000000) ? true : false;
+        bool is_naninf = ((val_bit_repr & fp16_inf) == fp16_inf) ? true : false;
+
+        int dshift = 0;
+        if (exp_h > 8) {  // too large, set it to NaN or inf
+            is_naninf = true;
+            if (use_clamp) {
+                exp_h = 8;
+                mantissa_h = 0b0000001100000000;
+            } else {
+                mantissa_h = 0;
+                exp_h = 16;
+            }
```

Comment:
This is dead code, remove?

Reply:
Keeping for now in case of a request for an adjustment of the logic (and update to use_clamp == false).

Reply:
I suggest leaving it. Because in the future we may move clamp outside of this operation.